src/main/java/com/puppycrawl/tools/checkstyle/checks/sizes/ExecutableStatementCountCheck.java

```diff
@@ -219,19 +225,18 @@ private void leaveMemberDef(DetailAST ast) {
      * @param ast the token representing the statement list.
      */
     private void visitSlist(DetailAST ast) {
-        if (context.getAST() != null) {
-            // find member AST for the statement list
-            final DetailAST contextAST = context.getAST();
-            DetailAST parent = ast.getParent();
-            int type = parent.getType();
-            while (type != TokenTypes.METHOD_DEF
-                && !isConstructorOrInit(type)) {
-                parent = parent.getParent();
-                type = parent.getType();
-            }
-            if (parent == contextAST) {
-                context.addCount(ast.getChildCount() / 2);
-            }
+        // find member AST for the statement list
+        final DetailAST contextAST = context.getAST();
+        DetailAST parent = ast.getParent();
+        int type = parent.getType();
+        while (type != TokenTypes.METHOD_DEF
```

Comment:
what do you think about moving these conditions to `isConstructorOrInit` and getting rid of `type` variable?

Reply:
And I would suggest to rename `isConstructorOrInit` method to more appropriate name

which name can you propose? :)

Reply:
I remember I wrote a utility for this. We can simply use that.